src/clj/rems/api/invitations.clj

```diff
@@ -0,0 +1,49 @@
+(ns rems.api.invitations
+  (:require [compojure.api.sweet :refer :all]
+            [rems.api.services.invitation :as invitation]
+            [rems.api.util :refer [not-found-json-response]] ; required for route :roles
+            [rems.application.events :as events]
+            [rems.common.roles :refer [+admin-read-roles+ +admin-write-roles+]]
+            [rems.schema-base :as schema-base]
+            [rems.util :refer [getx-user-id]]
+            [ring.util.http-response :refer :all]
+            [schema.core :as s])
+  (:import [org.joda.time DateTime]))
+
+(s/defschema CreateInvitationCommand
+  {:email s/Str
+   (s/optional-key :workflow-id) s/Int})
+
+(s/defschema CreateInvitationResponse
+  {:success s/Bool
+   (s/optional-key :id) s/Int
+   (s/optional-key :errors) [s/Any]})
+
+(s/defschema InvitationResponse
+  {(s/optional-key :invitation/id) s/Int
+   :invitation/email s/Str
+   :invitation/invited-by schema-base/UserWithAttributes
+   (s/optional-key :invitation/invited-user) schema-base/UserWithAttributes
+   :invitation/created DateTime
+   (s/optional-key :invitation/sent) DateTime
+   (s/optional-key :invitation/workflow) {:workflow/id s/Int}})
```

Comment:
Should the workflow be optional? What are the use cases if we do not us the workflow?

Reply:
If it's an invitation to something else, then there'll be something else. So far this does not replace the application invitations but an invitation to ownership has been discussed before (or any other role really).

Reply:
At this point we do not allow invites to anything else, and it is debatable whether we should allow invites to other things. But currently not.